### PR TITLE
fix: move and document the signature helpers 

### PR DESCRIPTION
**Pull request by grantseltzer**
This moves both the rego and go helper functions into their own package, updates usage and Makefile accordingly. (#599) 

It also adds notes in the README about them (#538).

Signed-off-by: grantseltzer <grantseltzer@gmail.com>

### CODE REVIEW

**Comment by grantseltzer** [comment]
Couple of issues I need to fix here:

- `findRegoSigs(dir string)` breaks because it's looking for `helpers.rego` at an essentially hardcoded path. I feel this should not be the case, or at the very least should be an optional thing to look for (helpers don't have to be used).
- A linter error

File: tracee-rules/signature_test.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ import (
 )
 
 func Test_getSignatures(t *testing.T) {
-	sigs, err := getSignatures("signatures/rego", []string{"TRC-2"})
+	sigs, err := getSignatures("./dist/rules", []string{"TRC-2"})
```
Comment (simar7) [inline tracee-rules/signature_test.go:13]: Passing `signatures/rego` makes the use of existing rego signatures as inputs, effectively using them as golden files for this test.

Was there another motivation to change it to `dist/rules`?

Comment (itaysk) [reply tracee-rules/signature_test.go:13]: I think it's ok to leave the helpers under the signatures directory, after all they are signature helpers. that will also address this concern I believe

**Review by itaysk — COMMENTED** [review]
I do think the rego helpers should move to a dedicated package (just the name, not the directory). in rego conventions that would be like `package tracee.helpers`

**Comment by itaysk** [comment]
also, probably worth adding code comments to document the helper functions themselves. but this can be a separate PR if you'd like

**Comment by grantseltzer** [comment]
> I do think the rego helpers should move to a dedicated package (like you already did before) (just the name, not the directory). in rego conventions that would be like `package tracee.helpers`

How is this different from how it is now?

> also, probably worth adding code comments to document the helper functions themselves. but this can be a separate PR if you'd like

On it

**Comment by itaysk** [comment]
> How is this different from how it is now?

my bad :) got confused for a moment

**Review by itaysk — APPROVED** [review]

**Pull request merged**